command/agent/template/template.go

```diff
@@ -196,12 +198,19 @@ func (ts *Server) Run(ctx context.Context, incoming chan string, templates []*ct
 				// return. The deferred closing of the DoneCh will allow agent to
 				// continue with closing down
 				ts.runner.Stop()
-				return
+				return nil
 			}
 		}
 	}
 }
 
+func (ts *Server) Stop() {
+	if !ts.stopped.Load() {
```

Comment:
You probably want a CompareAndSwap here to ensure we only call close once.

Reply:
Good call, done!